File: 3way/problem15/solution.c
int longestOnes(int* nums, int numsSize, int k) {
    int left = 0;
    int right = 0;
    int zero_count = 0;
    int max_value = 0;

    while(right < numsSize){
        if(nums[right] == 0){
            zero_count++;
        }

        while(zero_count > k){
            if(nums[left] == 0){
                zero_count--;
            }
            left++;
        }

        max_value = max_value > (right - left + 1) ? max_value : right - left + 1;

        right++;
    }

    return max_value;
}

File: 3way/problem15/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var longestOnes = function(nums, k) {
    let left = 0, right = 0;
    let zeroCount = 0;
    let max = 0;
    while(right < nums.length){
        if(nums[right] == 0){
            zeroCount++;
        }

        while(zeroCount > k){
            if(nums[left] == 0){
                zeroCount--;
            }
            left++;
        }

        max = Math.max(max, right - left + 1);

        right++;
    }
    return max;
};
